### ISSUE TITLE
Inadequate Testing Measures

### ISSUE THREAD

**Issue by ta2edchimp** [kentcdodds/eslint-find-new-rules#19]
We somehow stumble over an issue with our testing of `./bin.js`.

Currently in `./test/bin.js` the stub for proxyquire completely omit's the `findNewRules` functionality and always returns the array specified within a particular test:

``` JavaScript
// test/bin.js

const indexStub = {
  './index': () => rules,
}

// ...

test('new rule and no path', () => {
  rules = ['foo', 'bar', 'baz']
  process.cwd = () => noSpecifiedFile
  proxyquire('../bin', indexStub)
})
```

It is completely irrelevant what exactly happens within `../bin` because the test is assumed to pass, when `bin`'s response equals to the `rules` array:

``` JavaScript
function outputStatement() {
  return `New rules to add to the config: ${rules.join(', ')}.`
}

test.beforeEach((t) => {
  process.argv.splice(2, process.argv.length)
  process.exit = status => t.same(status, 1)
  console.log = statement => t.same(statement, outputStatement()) // eslint-disable-line no-console
})
```

I propose, we drop the mocking of `./index` in the proxyquire stab and rather mock the `readdirSync` method as we do in `./test/index.js`.
That way, we can test more thoroughly towards the rules that really have been specified within our fixtures ([I set up a branch for testing purposes](https://github.com/ta2edchimp/eslint-find-new-rules/blob/test-issues/test/bin.js)). That way we'd test the real functionality ...

**Comment by ta2edchimp** [comment]
**TL;DR:**

We're currently testing `rules` to be equal to `rules` :smile:

**Comment by ta2edchimp** [comment]
Self-assigned this issue.
Will rework the testing according to [this](https://github.com/kentcdodds/eslint-find-new-rules/pull/18#issuecomment-198984786).
